docs/operations/gotchas.md

```diff
@@ -0,0 +1,306 @@
+- Use read-only Redis users for the agent (ACLs with `+@read -@write -@dangerous`)
+- Deploy agent as a sidecar to the database and inject connection URL at runtime (requires custom scripting currently)
+- Monitor agent actions via OpenTelemetry traces and logs
+- Use network policies to restrict agent connectivity
+
+**Future direction**: Better patterns for sidecar deployment and runtime credential injection are being explored. The correct design is still simmering.
+
+### Triage sessions can be slow (7-10 minutes)
+The agent's deep-research approach is thorough but slow. A single triage session can take 7-10 minutes due to:
+- **Deep research**: Parallel research tracks with multiple tool-calling loops
+- **GPT-4o**: Default model choice (high quality, but slower than GPT-3.5)
+- **Large context**: Metrics and log data can be substantial (thousands of lines)
+- **Multiple LLM calls**: Each research track makes 1-2 generation calls, often 1-2 minutes each
+
+**The tradeoff**: Slow but valuable. The agent produces thorough analysis with actionable recommendations.
+
+**Workarounds**:
+- Use a faster model: Set `OPENAI_MODEL=gpt-3.5-turbo` (lower quality, faster)
+- Reduce max iterations: Set `MAX_ITERATIONS=15` (less thorough, faster)
+- Limit tool timeouts: Set `TOOL_TIMEOUT=30` (less data gathering, faster)
+- Use schedules for non-urgent checks (run overnight, review results in the morning)
+
+**Future direction**: We're exploring ways to make "intensity" configurable (think harder vs. don't think too much). Options under consideration:
+- Query-time intensity parameter (quick scan vs. deep dive)
+- Streaming partial results (show findings as they arrive)
+- Smarter context pruning (send less data to LLM without losing signal)
+
+---
+
+## General
+
+### Docker-compose is not for production
+The `docker-compose.yml` setup is for local development and testing only. It includes:
+- Demo Redis instances
+- Full observability stack (Prometheus, Grafana, Loki, Tempo)
+- Redis Enterprise test cluster
+
+For production, deploy the agent on VMs or in a container orchestration system and connect to your existing observability infrastructure.
+
+### Redis Enterprise vs Redis OSS
+The agent requires Redis with the RediSearch module for vector search. Options:
+- **Redis Enterprise**: RediSearch included, recommended for production
+- **Redis Stack**: RediSearch included, good for development
+- **Redis OSS**: Requires manual RediSearch module installation
+
+Redis 8.x is recommended. Redis 7.x works but may have different module versions.
+
+---
+
+## Configuration
+
+### Master key is required for production
+The `REDIS_SRE_MASTER_KEY` is used for envelope encryption of secrets (connection URLs, passwords, API keys). Without it:
+- Secrets are stored in plaintext in Redis
+- You'll see warnings in logs
+- Not suitable for production
```

Comment:
There isn't really another option, so let's strike this, as the agent is an early release that generally isn't ready for production.